ci/full-dependencies/Dockerfile

```diff
@@ -0,0 +1,103 @@
+# PANOPTES development container
+# Build with:
+#
+#         docker build --tag panoptes/build --file $POCS/ci/Dockerfile-3 -- $POCS
+#
+# This makes $POCS the "context" for the build, i.e. the root folder
+# from where files such as requirements.txt may be copied.
+
+FROM ubuntu:bionic as build-env
+
+LABEL description="PANOPTES Development Container"
+LABEL author="Developers for PANOPTES project"
+LABEL url="https://github.com/panoptes/POCS"
+
+ARG pan_dir=/var/panoptes
+
+ENV LANG=C.UTF-8 LC_ALL=C.UTF-8
+ENV ENV /root/.bashrc
+ENV SHELL /bin/bash
+ENV PANDIR $pan_dir
+ENV PANLOG $PANDIR/logs 
+ENV POCS $PANDIR/POCS  
+ENV PAWS $PANDIR/PAWS  
+ENV PANUSER root
+ENV SOLVE_FIELD=/usr/bin/solve-field
+
+################################################################################
+# Several update steps before our custom commands. By rarely changing these
+# commands, we allow the caching that 'docker build' performs to save us some
+# time when iterating on the Panoptes specific steps (i.e. instead of executing
+# these, the cached image layer is used).
+
+# Update the information we know about package versions.
+RUN apt-get update --fix-missing
+
+# Install the tzdata package now with interactive prompting disabled.
+# Conveniently, the default timezone is UTC.
+RUN DEBIAN_FRONTEND=noninteractive apt-get install --yes --quiet tzdata < /dev/null
+
+# Upgrade other packages.
+RUN DEBIAN_FRONTEND=noninteractive apt-get upgrade --yes --quiet < /dev/null
+
+
+
+################################################################################
+# DO NOT KEEP THIS LONG TERM!
+# Just using during editing and testing of install-dependencies.sh.
+
+WORKDIR $POCS/scripts
+COPY scripts/install install
+RUN install/install-apt-packages.sh
+WORKDIR $POCS
+RUN rm -rf scripts
+
+################################################################################
+
+# Make sure we have the appropriate bash version.
+RUN rm /bin/sh \
+ && ln -s /bin/bash /bin/sh
+
+WORKDIR $POCS
+
+# Copy POCS sources (not git repository) into this temporary directory.
+#
+# TODO(jamessynge): Use a command like the following to generate a list
+# of files/directories in POCS to copy, and a script that inserts that
+# into here for copying.
+#
+#         git ls-tree HEAD | cut -d$'\t' -f2 | xargs -i echo "COPY {} {}"
+# 
+# This command produces a single line version, which I've hand wrapped below.
+#
+#         echo COPY $(git ls-tree HEAD | cut -d$'\t' -f2) '$POCS/'
+
+# First we copy the files and directories least likely to change while
+# iterating on the building of the Docker image.
+COPY .codecov.yml .coveragerc .gitignore .gitmodules .pycodestyle.cfg \
```

Comment:
I don't understand why the selective copying? Is this also for caching purposes?